VmAgent.Core/Dependencies/BasicAssetExtractor.cs

```diff
@@ -0,0 +1,108 @@
+﻿using Microsoft.Azure.Gaming.VmAgent.Core.Dependencies.Interfaces;
+using Microsoft.Azure.Gaming.VmAgent.Core.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+using VmAgent.Core.Interfaces;
+
+namespace Microsoft.Azure.Gaming.VmAgent.Core.Dependencies
+{
+    public class BasicAssetExtractor: IBasicAssetExtractor
+    {
+        public static BasicAssetExtractor Instance = new BasicAssetExtractor();
```

Comment:
Singletons should not have a public setter, which this has. this should be defined as :
public static BasicAssetExtractor Instance {get;} = new BasicAssetExtractor();

Reply:
Thanks for the feedback! I'm not sure if we can access globals from BasicAssetExtractors considering Globals.cs is under LocalMultiplayerAgent and BasicAssetExtractos is under VmAgent.Core.

Reply:
Perhaps we can remove the logger for now and log that line while calling the ExtractAssets method. IT is not the greatest solution, but gives a way out for now. As for why not a static class, it needs input from outside to set up the loggers within it. An in general static classes have a very small set of specific use cases that we should use them for.

Reply:
What about this approach? Would it be not recommended?

public static BasicAssetExtractor Instance { get; } = new BasicAssetExtractor();

        private readonly ISystemOperations _systemOperations;
        private readonly MultiLogger _logger;

        public BasicAssetExtractor(ISystemOperations systemOperations = null, MultiLogger logger = null)
        {
            _systemOperations = systemOperations ?? SystemOperations.Default;
            _logger = logger;
        }

Reply:
You have to still instantiate Instance outside of this class right? like you are doing it in the Program.cs. IF that's the case, we should remove the Instance completely. You can then just create a new instance wherever you want rather than trying to use a singleton Instance. It is perhaps better that way.

Reply:
I think if we remove the logger and make sure SystemOperations.Default is initialized before the instance is used, we don't need to instantiate outside.  Another way as you suggested, I can also remove instance completely and create a new instance like code below. This case we can keep loggers. If I missed the point, please correct me! 

public MultiplayerServerManager(
            ISystemOperations systemOperations,
            VmConfiguration vmConfiguration,
            MultiLogger logger,
            ISessionHostRunnerFactory sessionHostRunnerFactory,
            BasicAssetExtractor basicAssetExtractor = null )
        {
            _sessionHostRunnerFactory = sessionHostRunnerFactory;
            _systemOperations = systemOperations;
            _logger = logger;
            _vmConfiguration = vmConfiguration;
            _basicAssetExtractor = new basicAssetExtractor(_systemOperations, _logger);
        }

Reply:
The code above seems fine to me. You can keep the logger and remove the Instance

Reply:
updated